dolphin/alert_manager/trap_receiver.py

```diff
@@ -43,6 +45,102 @@ def __init__(self, trap_receiver_address, trap_receiver_port,
         self.trap_receiver_port = trap_receiver_port
         self.snmp_mib_path = snmp_mib_path
 
+    def sync_snmp_config(self, ctxt, snmp_config_to_del=None,
+                         snmp_config_to_add=None):
+        if snmp_config_to_del is not None:
+            self._delete_snmp_config(ctxt, snmp_config_to_del)
+
+        if snmp_config_to_add is not None:
+            self._add_snmp_config(ctxt, snmp_config_to_add)
+
+    def _add_snmp_config(self, ctxt, new_config):
+        LOG.info("Add snmp config:%s" % new_config)
+        storage_id = new_config["storage_id"]
+        version_int = self._get_snmp_version_int(ctxt, new_config["version"])
+        if version_int == constants.SNMP_V2_INT or \
+                version_int == constants.SNMP_V1_INT:
+            community_string = new_config["community_string"]
+            community_index = self._get_community_index(storage_id)
+            config.addV1System(self.snmp_engine, community_index,
+                               community_string, contextName=community_string)
+        else:
+            username = new_config["username"]
```

Comment:
new_config is dict? then why to use index at few places and get at some places? I think you need to use get everywhere, or handle KeyError exception too

Reply:
You can omit None..It's the default for get()

Reply:
Done